### ISSUE TITLE
Distortion in Object rotation

### ISSUE THREAD

**Issue by oliveira500** [Samsung/GearVRf#160]
Using the method getTransform().setRotation(w,x,y,z) is occurring distortion of the object . 

![distortion](https://cloud.githubusercontent.com/assets/4211447/8708040/9fba747c-2b07-11e5-85a8-73ecb41a7fe2.png)
![object](https://cloud.githubusercontent.com/assets/4211447/8708041/9fccb11e-2b07-11e5-9ba1-d4c29a17d667.png)

**Comment by phi-lira** [comment]
Are you passing in an normalized vector (x, y, z) to setRotation? Can you share the values?

**Comment by oliveira500** [comment]
Yes, I'm passing a normalized vector. This bug does not happen with simple objects, It's occurring the object, which is composed of multiple polygons.
Can I share the values, object and texture.

**Comment by chenchao1407** [comment]
Can you send me the whole project package to me about the bug? Thanks in advance.
My email address is c.chen@samsung.com

**Comment by oliveira500** [comment]
I thank you attention. I'll send now.

**Comment by phi-lira** [comment]
I got the project and tested here. 

I think this is more of a SDK misleading interface problem than a bug. setRotation does not work as you expect. It takes a raw quaternion instead and passing an angle/axis will produce wrong composition of rotation/scale matrix.

IMO, we should make obsolete setRotation(w, x, y, z) as this is error prone. We should also add interface to setRotationByEulerAngles( yaw, pitch, roll). 

If one wants to pass in a quaternion we should create a GVRQuaternion helper class that can construct quaternions from eulerangles, angleAxis, two vectors, but never from raw w, x, y, z. Interface would be setRotation(GVRQuaternion quat)

**Comment by thomasflynn** [comment]
Right, the GVRTransform api seems to have been written assuming using quaternions would be the more frequent use case.  For the traditional glRotatef() type functions, you're expected to use rotateByAxis() method.  Though, note, even if you do, there's no equivalent to glLoadIdentity() that i saw.  GVRTransform could certainly be cleaned up a bit more.  I will likely add a loadIdentity() method tonight, 'cause I need it to continue porting a project i wrote several years ago.  A GVRQuaternion class would be a welcome contribution by someone :)

**Comment by phi-lira** [comment]
I think it's good to keep GVRTransform.rotation as a quaternion. We should just make sure client code don't have access to set it to values that don't make sense (like setRotation currently does). 

I agree a loadIdentity would be really helpful. I think calling it reset() would make it even more user friendly as not everyone is used to the matrix identity jargon.

I'll create feature request for GVRQuaternion/GVRVector.
